From 523638f45e7c65f2ee447f1094fedb945fd8cec9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Thu, 8 Sep 2022 15:28:02 +0300
Subject: [PATCH 83/85] contrib: modplug: Force building with a specific older
 C++ version

This project uses the 'register' storage class specifier for a lot
of variables, and C++17 disallows that storage class spcifier.

This fixes building with newer compilers that default to C++17
if nothing is specified (which is the default since GCC 11
and Clang 16).

When building with Clang 16 in C++17 mode, it fails with errors
like these:

fastmix.cpp:678:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
        SNDMIX_BEGINSAMPLELOOP8
        ^
fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
        register MODCHANNEL * const pChn = pChannel;\
        ^
---
 contrib/src/modplug/rules.mak | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/contrib/src/modplug/rules.mak b/contrib/src/modplug/rules.mak
index aa10fc41bb..c8371c9270 100644
--- a/contrib/src/modplug/rules.mak
+++ b/contrib/src/modplug/rules.mak
@@ -8,6 +8,8 @@ ifeq ($(call need_pkg,"libmodplug >= 0.8.9.0"),)
 PKGS_FOUND += modplug
 endif
 
+MODPLUG_CXXFLAGS := $(CXXFLAGS) -std=gnu++98
+
 $(TARBALLS)/libmodplug-$(MODPLUG_VERSION).tar.gz:
 	$(call download_pkg,$(MODPLUG_URL),modplug)
 
@@ -26,6 +28,6 @@ endif
 
 .modplug: libmodplug
 	$(RECONF)
-	cd $< && $(HOSTVARS) ./configure $(HOSTCONF)
+	cd $< && $(HOSTVARS) CXXFLAGS="$(MODPLUG_CXXFLAGS)" ./configure $(HOSTCONF)
 	cd $< && $(MAKE) install
 	touch $@
-- 
2.43.0

